### PR TITLE
feat: add secret and endpoint restrictions

### PR DESCRIPTION
**Pull request by julianiff**
-> adding unique constraint to keep Secret Key and Endpoint Name unique.

### CODE REVIEW

**Review by oliverschwendener — APPROVED** [review]
Looks good to me, though I have no clue why `verifyManifest.ts` is owned by us 🤷

File: packages/cli/src/utils/verifyManifest.ts [hunk 1]
```diff
@@ -18,11 +18,27 @@ const logoLibraryFilenameExtension = z.enum(['svg', 'jpg', 'jpeg', 'ai', 'eps',
...
+            if (secretKeySet.has(key)) {
+                return false;
+            }
+
+            secretKeySet.add(key);
+            return /^[\w-]+$/.test(key);
+        },
+        {
+            message:
+                "The key should only contain letters from a-z, A-Z, numbers from 0-9, '-' and '_' without any spaces",
```
Comment (fulopdaniel) [inline packages/cli/src/utils/verifyManifest.ts:39]: Does it make sense to add "key must be unique" to this error message?

Comment (imoutaharik) [reply packages/cli/src/utils/verifyManifest.ts:39]: I agree with @fulopdaniel , it would be nice to add it to the message.

Comment (julianiff) [reply packages/cli/src/utils/verifyManifest.ts]: great catch updated

**Review by fulopdaniel — APPROVED** [review]

**Review by imoutaharik — APPROVED** [review]
LGTM. :)

**Pull request merged**